alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt

```diff
@@ -525,12 +547,321 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
         assertEquals(4, adminResponseMap["totalAlerts"])
 
         // add alerting roles and search as userOne - must return 1 docs
-        createUserRolesMapping(ALERTING_GET_ALERTS_ACCESS, arrayOf(user))
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_GET_ALERTS_ACCESS)
+        )
         try {
             val responseMap = getAlerts(userClient as RestClient, inputMap).asMap()
             assertEquals(4, responseMap["totalAlerts"])
         } finally {
-            deleteRoleMapping(ALERTING_GET_ALERTS_ACCESS)
+            deleteRoleAndRoleMapping(TEST_HR_ROLE, ALERTING_GET_ALERTS_ACCESS)
+        }
+    }
+
+    // Email account related tests.
+
+    fun `test search email accounts with an user with search email account role`() {
```

Comment:
Email and destination related tests need to be in the `SecureDestinationRestApiIT.kt` file.

Reply:
moved.